core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -115,6 +115,8 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
         return UIUtils.headerSparkPage(request, stageHeader, content, parent)
       }
 
+    val stageJobIds = parent.store.getJobIdsAssociatedWithStage(stageId, stageAttemptId)
```

Comment:
In https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala#L109 there is a query for the stage data already. We can reduce the query to the store here.

Reply:
E.g. we can add a function to return the whole StageDataWrapper in AppStatusStore

Reply:
Makes sense, I have fixed it. Request you to have a look.